receiver/prometheusreceiver/internal/otlp_metricfamily.go

```diff
@@ -172,6 +172,23 @@ func (mg *metricGroupPdata) toSummaryPoint(orderedLabelKeys []string, dest *pdat
 	return true
 }
 
+func (mg *metricGroupPdata) toNumberDataPoint(orderedLabelKeys []string, dest *pdata.NumberDataPointSlice) bool {
+	var startTsNanos pdata.Timestamp
+	tsNanos := pdata.Timestamp(mg.ts * 1e6)
+	// gauge/undefined types have no start time.
+	if mg.family.isCumulativeTypePdata() {
+		startTsNanos = tsNanos
```

Comment:
This is not correct. The starttime should not be set to the current timestamp, but to the beginning of the interval.

Reply:
https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L132

Reply:
Got it, thank you @bogdandrutu! That means that the prior code was incorrect as this is a translation. I have filed this issue https://github.com/open-telemetry/opentelemetry-collector/issues/3691 as a bug

Reply:
@bogdandrutu please take a look, I've filed the appropriate bug and I've added TODOs to ensure that after this merge I shall send a follow up to fix that problem. Thank you!

Reply:
@bogdandrutu I've mailed out PR https://github.com/open-telemetry/opentelemetry-collector/pull/3694 which is based off this one, and after this one is merged, we'll rebase. Thank you